torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -4481,7 +4482,7 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::barrier(const BarrierOptions& opts) {
       at::zeros({1}, at::TensorOptions().device(barDevice).dtype(at::kFloat));
 
   // All reduce to achieve the barrier
-  auto work = allreduce_impl(barrierTensor);
+  auto work = allreduce_impl<true>(barrierTensor);
```

Comment:
nit: does this work?  Its nice to avoid passing mysterious booleans without names in c++ using comments  `auto work = allreduce_impl</*isBarrier*/true>(barrierTensor);`

Reply:
Yeah, I am searching the test methods, would update when test results are ready.

Reply:
ProfilingTitle should be used in multiple places in UTs? add a barrier call there to verify? And you can check the github issue which originally complains about the profileTitle being confusing, and re-pro the same test?